src/routes/connect/ui/wallets/Wallet.ts

```diff
@@ -0,0 +1,38 @@
+import type IWallet from './interfaces/IWallet';
+import InvalidComponentTypeError from '../../../../lib/errors/InvalidComponentTypeError';
+import Albedo from './Albedo';
+import Freighter from './Freighter';
+import PrivateKey from './PrivateKey';
+import Rabet from './Rabet';
+import XBull from './XBull';
+
+export default class WalletFactory {
+    create(name: string, privateKey?: string): IWallet {
+        let wallet!: IWallet;
+        switch (name) {
+            case 'albedo':
+                wallet = new Albedo();
+                break;
+            case 'xbull':
+                wallet = new XBull();
+                break;
+            case 'rabet':
+                wallet = new Rabet();
+                break;
+            case 'freighter':
+                wallet = new Freighter();
+                break;
+            case 'privateKey':
+                wallet = new PrivateKey(privateKey!);
```

Comment:
Generally the `!` is a sign you should do something different. In this case, it'd be a requirement that if the case is privateKey that the privateKey is not null, in which case you can create a condition and throw an MissingPrivateKey exception or something similar.